File: 프로그래머스/1/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    answer = []
    dict_name = dict(zip(name, yearning))

    for i in range(len(photo)):
        sum_ = 0
        for j in photo[i]:
            # sum_ += dict_name[i][j]
            # print(sum_)
            if j in name:
                sum_ += dict_name[f"{j}"]
        answer.append(sum_)
    return answer

File: 프로그래머스/1/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
    let score = {}
    let answer = Array(photo.length).fill(0)
    for (let i = 0;i < name.length;i++) {
        score[name[i]] = yearning[i];
    }
    for (let i = 0;i < photo.length;i++) {
        photo[i].map((v) => {
            if(score[v] !== undefined) {
                answer[i] += score[v];
            }
        })
    }
    return answer;
}